stage0/run.go

```diff
@@ -314,7 +314,7 @@ func preparedWithPrivateUsers(dir string) (string, error) {
 
 // Run mounts the right overlay filesystems and actually runs the prepared
 // pod by exec()ing the stage1 init inside the pod filesystem.
-func Run(cfg RunConfig, dir string) {
+func Run(cfg RunConfig, dir string, dataDir string) {
```

Comment:
I am even less thrilled with adding flavor-specific (thus stage1 specific) stuff in stage0.

Is encryption here important? Or access control? Maybe we could consider using telnet? That probably would avoid a hassle with setting up keys on host side, right?